### PR TITLE
Page List : Add border and spacing support

### PR DESCRIPTION
**Pull request by benazeer-ben**
## What?

Adding block & spacing support for Page list block.

Part of https://github.com/WordPress/gutenberg/issues/43247 and  https://github.com/WordPress/gutenberg/issues/43243

## Why?

Page list block is missing border & spacing  support

## How?

Adds the border & spacing  support via block.json

## Testing Instructions

- Go to Global Styles setting ( under appearance > editor > styles > edit styles > blocks )
- Make sure that `Page List `block's border is configurable via Global Styles
- Verify that Global Styles are applied correctly in the editor and frontend
- Edit template/page, Add `Page List` block and apply the border styles & spacing (padding, margin)styles
- Verify that block styles take precedence over global styles
- Verify that block borders & spacing display correctly in both the editor and frontend

## Screenshots or screencast 

In Backend:
pagelistbackend

In Frontend:
pagelistfrontend

### CODE REVIEW

File: packages/block-library/src/page-list/block.json [hunk 1]
```diff
@@ -51,6 +51,18 @@
...
+			"radius": true,
+			"color": true,
+			"width": true,
+			"style": true,
+			"__experimentalDefaultControls": {
+				"radius": true,
+				"color": true,
+				"width": true,
+				"style": true
+			}
```
Comment (aaronrobertshaw) [inline packages/block-library/src/page-list/block.json]: These border controls should probably not be displayed by default given the block doesn't already have spacing support controls shown by default.

File: packages/block-library/src/page-list/block.json [hunk 2]
```diff
@@ -51,6 +51,18 @@
 		},
 		"interactivity": {
 			"clientNavigation": true
+		},
+		"__experimentalBorder": {
+			"radius": true,
```
Comment (aaronrobertshaw) [inline packages/block-library/src/page-list/block.json:56]: To properly support border radius, we likely need to also adopt padding and margin supports for the block.

**Review by aaronrobertshaw — COMMENTED** [review]
Thank you for putting together this PR @benazeer-ben 👍 

For the most part it's working pretty well for me. There are a few things we might need to tweak before landing it though.

✅ Global styles apply correctly to the block
✅ Block instance styles override global styles
✅ Box sizing allows parent to control width effectively while borders are present

❓ Default controls - There's a general direction at present that border controls are only displayed by default if the block already had spacing controls e.g. padding/margin, and the spacing controls were displayed by default.

❓ Lack of padding support - Adopting border radius can mean clipping content. Normally this is mitigated somewhat by the user also having control over padding on the block. We might need to expand the scope of this PR to also adopt spacing supports.

Here's what I see when testing this branch:

https://github.com/user-attachments/assets/0cfb9c94-2ca3-4882-95c2-25b08c9da521

Thanks again for tackling this, @benazeer-ben, appreciate it!

**Comment by benazeer-ben** [comment]
Hi @aaronrobertshaw, Thanks for your detailed feedback.

I have added padding and margin support for the page list block. Please have a look at it and share your thoughts.

File: packages/block-library/src/page-list/block.json (hunk 2 above)
Comment (aaronrobertshaw) [reply packages/block-library/src/page-list/block.json:56]: Resolved via https://github.com/WordPress/gutenberg/pull/66385/commits/3406a5ad551f538a8b8cb3220c9cb157451bcdaf however the spacing supports likely shouldn't be displayed by default.

File: packages/block-library/src/page-list/block.json (hunk 1 above)
Comment (aaronrobertshaw) [reply packages/block-library/src/page-list/block.json]:
```suggestion
```

These should be hidden by default given the block didn't already have spacing support and is getting it primarily for border-radius support and consistency.

File: packages/block-library/src/page-list/block.json [hunk 3]
```diff
@@ -51,6 +51,22 @@
...
+			"__experimentalDefaultControls": {
+				"radius": true,
+				"color": true,
+				"width": true,
+				"style": true
+			}
+		},
+		"spacing": {
+			"padding": true,
+			"margin": true
```
Comment (aaronrobertshaw) [inline packages/block-library/src/page-list/block.json]: As noted above, we should probably hide these by default here.
```suggestion
			"padding": true,
			"margin": true,
			"__experimentalDefaultControls": {
				"padding": false,
				"margin": false
			}
```

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for iterating on this one @benazeer-ben 👍 

It's very close. All that is missing is to tweak which block support controls are displayed by default.

✅ Global styles apply correctly to the block
✅ Block instance styles override global styles
✅ Box sizing allows parent to control width effectively while borders are present
❓ Controls still need to be hidden by default

Given the Page List block didn't previously have spacing support and is only getting it due to the border radius support, all the controls should be optional i.e. hidden by default.

I've left a couple of tweaks via inline comments. With those applied this is working as expected for me.

Screenshot 2024-11-11 at 3 41 54 pm

**Review by aaronrobertshaw — COMMENTED** [review]
Loving the quick turnaround on iterations here @benazeer-ben ⚡ 

I was giving this a final test and noticed that the `box-sizing: border-box` CSS is only applied in the editor. We'll need to add it to the blocks styles so parent blocks can still enforce a max width on Page List blocks.

Here's a quick diff you can apply to get the job done:

```diff
diff --git a/packages/block-library/src/page-list/style.scss b/packages/block-library/src/page-list/style.scss
index d06a614257..a77fb85903 100644
--- a/packages/block-library/src/page-list/style.scss
+++ b/packages/block-library/src/page-list/style.scss
@@ -15,3 +15,9 @@
 		background-color: inherit;
 	}
 }
+
+.wp-block-page-list {
+	// This block has customizable padding, border-box makes that more predictable.
+	box-sizing: border-box;
+}
+

```

To test this:
1. Add a Page List block and apply both padding and a border
2. Wrap it in a Group block
3. Set a content width like `300px` on the Group block
4. Save and switch to the frontend
5. Note that the Page List should now respect the `300px` width applied by the parent Group block.

**Comment by benazeer-ben** [comment]
Thanks again for your detailed feedback @aaronrobertshaw . `border-box` styling added in the last commit.

**Comment by benazeer-ben** [comment]
Hi @aaronrobertshaw Kindly let me know if there is any more feedback.
Thanks.

**Review by ramonjd — APPROVED** [review]
This is now working for me now after @aaronrobertshaw's suggestion, and that `box-sizing: border-box;` is applied in styles.scss.

I checked:

- theme.json block styles
- global styles in the site editor for page list block
- block styles

All are working for me.

**Review by aaronrobertshaw — APPROVED** [review]
LGTM as well ✨ 

✅ Global styles are applied correctly in editor and on frontend
✅ Block support styles override global styles appropriately
✅ Box sizing allows parent block to enforce width

I can confirm the issue from my last review has been resolved. After wrapping a page list in a group block and setting its content width to 250px, this is what I get:

Screenshot 2024-11-18 at 6 09 53 pm

**Pull request merged**